### ISSUE TITLE
Watch folder subfolders don't work

### ISSUE THREAD

**Issue by themightykitten** [rakshasa/rtorrent#833]
Hi,

I just finished setting everything up in my rtorrent.rc and I realize that the watch folder only works when .torrent files are placed at its root. If I put subfolders, the .torrent files in them never get added

Isn't it supposed to pick them and use the path (relative to the root of the watch folder) to set the path of the downloaded file (relative to the downloads folder) ?

Another question, when download finishes, the .torrent file is supposed to be deleted, right? It then only survives in the session folder, no?

Here's the bit of my rtorrent.rc

```
schedule2 = watch_directory,5,5,load.start=/watch/*.torrent
schedule2 = tied_directory,6,5,start_tied=/watch/*.torrent
schedule2 = untied_directory,7,5,stop_untied=/watch/*.torrent
```

Thanks

**Comment by pyroscope** [comment]
Use the  new inotify watch from 097.

**Comment by mohkale** [comment]
@pyroscope I'm running on WSL ubuntu with rtorrent 0.9.7, the inotify system still isn't working in subdirectories for me.

I've got the following in my rtorrentrc

```
directory.watch.added = (cfg.watch), load.verbose
directory.watch.added = (cfg.watch_start), load.start_verbose
```

`cfg.watch` is a path at `~/downloads/torrents/`.

If I add a torrent to `~/downloads/torrents/a.torrent`, rtorrent picks up on it. If I add a torrent to `~/downloads/torrents/foo/a.torrent` rtorrent doesn't pick up on it. Am I doing something wrong here?

**Comment by pyroscope** [comment]
What is the actual value of `(cfg.watch_start)`? Also WSL, who knows what weirdness that brings to the table.

**Comment by mohkale** [comment]
> What is the actual value of (cfg.watch_start)?

`(cfg.watch_start)` is just another directory, it's technically a subdirectory of `(cfg.watch)` but I tried removing it and that didn't make any difference. It's value is `~/downloads/torrents/.start`.

> WSL, who knows what weirdness that brings to the table.

👍. I'm migrating to arch today so I'll try on there and update when I can.

**Comment by Penguin-Guru** [comment]
I am having the same issue with inotify on native Linux. Files dropped directly into the specified directories are detected; not if they are dropped into a subdirectory. I tried upgrading from 0.15.5 to 0.16.5 but the behaviour did not change. I verified that inotify works on my system with `inotifywait -r`. Has anyone verified that subdirectories are supported by this implementation? I will try taking a look at the code sometime but that will probably not be any time soon.
